File: sbe/okx/okx_sbe_1_0/generated-python/com_okx_sbe/MessageHeader.py
"""Generated SBE (Simple Binary Encoding) message codec."""

import struct
from typing import List, Optional, Tuple
from io import BytesIO

class MessageHeader:
    """Composite type."""

    def __init__(self):
        self.block_length = 0
        self.template_id = 0
        self.schema_id = 0
        self.version = 0

    def encode(self, buffer: BytesIO) -> None:
        """Encode the composite to the buffer."""
        buffer.write(struct.pack('<H', self.block_length))
        buffer.write(struct.pack('<H', self.template_id))
        buffer.write(struct.pack('<H', self.schema_id))
        buffer.write(struct.pack('<H', self.version))

    def decode(self, buffer: BytesIO) -> None:
        """Decode the composite from the buffer."""
        self.block_length = struct.unpack('<H', buffer.read(2))[0]
        self.template_id = struct.unpack('<H', buffer.read(2))[0]
        self.schema_id = struct.unpack('<H', buffer.read(2))[0]
        self.version = struct.unpack('<H', buffer.read(2))[0]

    @staticmethod
    def encoded_length() -> int:
        return 8

File: sbe/binance/spot_3_1/generated-php/spot_sbe/MessageHeader.php
<?php
/**
 * Generated SBE (Simple Binary Encoding) message codec.
 */

class MessageHeader
{
    public int|float $blockLength = 0;
    public int|float $templateId = 0;
    public int|float $schemaId = 0;
    public int|float $version = 0;

    public function encode(): string
    {
        $buffer = '';

        $buffer .= pack('v', $this->blockLength);
        $buffer .= pack('v', $this->templateId);
        $buffer .= pack('v', $this->schemaId);
        $buffer .= pack('v', $this->version);

        return $buffer;
    }

    public function decode(string $data, int &$offset = 0): void
    {
        $this->blockLength = unpack('v', substr($data, $offset, 2))[1];
        $offset += 2;
        $this->templateId = unpack('v', substr($data, $offset, 2))[1];
        $offset += 2;
        $this->schemaId = unpack('v', substr($data, $offset, 2))[1];
        $offset += 2;
        $this->version = unpack('v', substr($data, $offset, 2))[1];
        $offset += 2;
    }

    public static function encodedLength(): int
    {
        return 8;
    }
}
